### PR TITLE
mautrix-telegram: 0.9.0 -> 0.10.0

### PR DESCRIPTION
**Pull request by Ma27**
###### Motivation for this change

ChangeLog: https://github.com/tulir/mautrix-telegram/releases/tag/v0.10.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 126872` at 3ca4e7a5 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/038fe726859a45dd5947728d82d2d720">mautrix-signal</a></li>
  </ul>
</details>
<details>
  <summary>5 packages built successfully:</summary>
  <ul>
    <li>mautrix-telegram</li>
    <li>python38Packages.mautrix (python38Packages.mautrix-appservice)</li>
    <li>python38Packages.telethon</li>
    <li>python39Packages.mautrix (python39Packages.mautrix-appservice)</li>
    <li>python39Packages.telethon</li>
  </ul>
</details>
<details>
  <summary>5 suggestions:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `async_generator`.
Near pkgs/development/python-modules/telethon/default.nix:1:48:
```
  |
1 | { lib, buildPythonPackage, fetchPypi, openssl, async_generator, rsa, pyaes, pythonOlder }:
  |                                                ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/servers/mautrix-telegram/default.nix:70:0:
```
   |
70 |     description = "A Matrix-Telegram hybrid puppeting/relaybot bridge";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/servers/mautrix-telegram/default.nix:24:15:
```
   |
24 |   patches = [ ./0001-Re-add-entrypoint.patch ./0002-Don-t-depend-on-pytest-runner.patch ];
   |               ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/servers/mautrix-telegram/default.nix:24:46:
```
   |
24 |   patches = [ ./0001-Re-add-entrypoint.patch ./0002-Don-t-depend-on-pytest-runner.patch ];
   |                                              ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/telethon/default.nix:30:0:
```
   |
30 |     description = "Full-featured Telegram client library for Python 3";
   | ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by Ma27** [comment]
Oh I wasn't aware of mautrix-signal... I guess we'll just keep it open then.

**Comment by SuperSandro2000** [comment]
> Oh I wasn't aware of mautrix-signal... I guess we'll just keep it open then.

Maybe we can widen the version range if it still works with the new one?

**Comment by Ma27** [comment]
I neither use signal nor mautrix-signal, so I don't have a reasonable way to test it. But is there anything wrong with adding `mautrix_0_8` & `mautrix_0_9` here for now and remove `_0_8` when the new mautrix-signal is out? These are applications that are incompatible with the other version, so a mix-up seems rather unlikely to me or am I missing some other issue with this approach?

**Comment by dotlambda** [comment]
Mautrix-signal should override the mautrix version.
This needs to be merged however because we have a version of ruamel.yaml in python-unstable that's only supported by mautrix-telegram 0.10.0.

**Comment by dotlambda** [comment]
Can someone who uses mautrix-signal please test this?
cc @expipiplus1

**Comment by dotlambda** [comment]
Result of `nixpkgs-review pr 126872` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>mautrix-signal</li>
    <li>mautrix-telegram</li>
    <li>python38Packages.mautrix</li>
    <li>python38Packages.telethon</li>
    <li>python39Packages.mautrix</li>
    <li>python39Packages.telethon</li>
  </ul>
</details>

File: pkgs/servers/mautrix-signal/default.nix [hunk 1]
```diff
@@ -1,6 +1,18 @@
-{ lib, python3Packages, fetchFromGitHub }:
+{ lib, python3, fetchFromGitHub }:
 
-python3Packages.buildPythonPackage rec {
+let
+  python = python3.override {
+    packageOverrides = self: super: {
+      mautrix = super.mautrix.overridePythonAttrs (oldAttrs: rec {
+        version = "0.8.18";
```
Comment (petabyteboy) [inline pkgs/servers/mautrix-signal/default.nix:7]: Same here: Please override name as well

Comment (petabyteboy) [reply pkgs/servers/mautrix-signal/default.nix:7]: Oops, overridePythonAttrs seems to fix this

**Pull request merged**